Issue 3672 - Add mechanism to update transition attributes on change transition obj

Fybex opened the pull request:
# Description

Changes:
- created `SVGTransitionMigrator` which checks `transtion.block` svg-icon-maxi attributes and modify them if it's needed
- renamed `transitionMigrator` to `buttonIconTransitionMigrator`
- created `getTransitionData` helper, which returns `data.transition` by block name
- fixed problem with tabs in transitions (always had only one setting - canvas or block after #3712)
 
<!---
Please include a summary of the changes and the related issue.
Please also include relevant motivation and context.
--->

Fixes #3672 

# How Has This Been Tested?
Checked if there was no transition in master with this [code editor](https://github.com/maxi-blocks/maxi-blocks/files/9717901/svg-with-old-transition.txt) didn't have transition, checked if the svg colour transition works in this branch. Checked if transition tabs working correct (block and canvas tabs, which appears in blocks with block and canvas settings on turning off change all switch)
<!---
Please describe the tests that you ran to verify your changes.
Provide instructions so we can reproduce.
Please also list any relevant details for your test configuration.
--->

# Test checklist

<!--- Please remove the unnecessary checkbox --->

**_ Front/Back Testing _**

-   [ ] Test the code editor (described how above in "How Has This Been Tested?")
-   [x] Test transition tabs correct work

**_ Pre-Code Testing _**

-   [ ] Test the code editor (described how above in "How Has This Been Tested?")
-   [ ] Test transition tabs correct work
-   [ ] Check no commented code and no unnecessary imports
-   [ ] Standards of the project have been followed
-   [ ] No errors/warnings on console

# Checklist

-   [X] My code follows the style guidelines of this project
-   [X] I have performed a self-review of my code
-   [X] My changes generate no new warnings/errors

Fybex committed:
`getTransitionData` func helper

Fybex committed:
`addNewTransitions` checked in maxiBlockComponent

Fybex committed:
Fix problem with displaying transition tabs(canvas, block)

Lol, why I was checking length of obj :-1:

Fybex committed:
Rerender block if transition has changed

Fybex committed:
Refactor `getFirstValueFromObject`

Fybex committed:
Mock `getTransitionData`

Naaaaiix reviewed (commented):
Let's see if I understood correctly what has been implemented here 😸 

When rendering a block, when `isMounted`, we iterate over transitions and ensure their structure is correct comparing with the default element, right? The idea is correct, but opens some thoughts I'd like to share:
1. This seems more a `migrator` than a method inside the block. The reason is, on one side organization (seems more a migrator IMO), and the other the capacity to be under a `isElegible` condition that may reduce the time of rendering the block. This implementation will iterate on an object, so 100 blocks will do the same when downloading from Library, and it can (and probably will) slow down the speed of rendering. We might ensure we reduce it using `isElegible`.
2. This kind of `checker` (this implementation checks the correct structure of `transitions` object attribute) could be applied for `relations` or even `background-layers` or `transform`. So, I'm not sure if it's a good idea implementing something like this branch propose, as is an enormous amount of process that for the moment we are applying just for `transitions`, but the same can be done for other `object` attributes and that can be a painful point for our speed. So, maybe we need to just fix this issue as a new migrator focused on adding the SVG transition, and don't check `transitions` structure every time we render a block.

Basically, I'm afraid of the speed reducing effect that this implementation can generate, so maybe we need to simplify and reduce as a simple migrator 🤔

Fybex committed:
`addNewTransition` code to `SVGTransitionMigrator`

Call migrator only for svg-icon-maxi, made it specified only for transition block to maximally optimize

Fybex committed:
Rename `transitionMigrator` to `buttonIconTransitionMigrator`

Naaaaiix reviewed (commented):
Awesome! 👏 

Need to be adapted to the new blockMigrator, but the essence is working perfect 👍

Fybex committed:
Merge branch 'master' into issue-3672

Fybex committed:
Update SVG Transition Migrator

Fybex committed:
Fix SVG Transition Migrator `isEligible`

Fybex committed:
Fix migrators attributes without attributes func

Fybex committed:
Don't run the migrator if !blockDataTransition
